### DISCUSSION TITLE
/lib/firmware/nvidia/tegra194/* files conflict with RootFS build

### DISCUSSION THREAD

**Discussion by andregirard37** [OE4T/meta-tegra#1496] [Q&A]
Hello,
I am working on a custom Yocto development build (using Kirkstone) and near the end of the build, I get the following error:

ERROR: nx-image do_rootfs: Could not invoke dnf. Command ... returned 1:
DNF version: 4.11.1
cachedir: /.../build/tmp/work/jetson_xavier_nx_devkit_emmc-poky-linux/nx-image-devel/1.0-r0/rootfs/var/cache/dnf
Added oe-repo repo from /.../build/tmp/work/jetson_xavier_nx_devkit_emmc-poky-linux/nx-image/1.0-r0/oe-rootfs-repo
User-Agent: falling back to 'libdnf': could not detect OS or basearch
repo: using cache for: oe-repo
oe-repo: using metadata from Wed 06 Mar 2024 03:19:53 PM UTC.

Long list of packages:
---> Package

Dependencies resolved.
...
Install  1624 Packages

Total size: 2.5 G
Installed size: 7.3 G
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
**Error: Transaction test error:
  file /lib/firmware/nvidia/tegra194/vic.bin conflicts between attempted installs of linux-firmware-nvidia-tegra-1:20231030-r0.noarch and tegra-firmware-tegra194-35.4.1+20230801124926-r0.armv8a_tegra194
  file /lib/firmware/nvidia/tegra194/xusb.bin conflicts between attempted installs of tegra-firmware-tegra194-xusb-35.4.1+20230801124926-r0.armv8a_tegra194 and linux-firmware-nvidia-tegra-1:20231030-r0.noarch**

I have no idea why it would look at /lib/firmware/nvidia/tegra194 in the first place. Is there something I need to update/upgrade ? That build used to work just fine on Dunfell 4.6.4.

Any help would be appreciated. Thanks!

**Comment by madisongh** [comment]
Assuming you're not explicitly including `linux-firmware-nvidia-tegra` in your image (which you shouldn't - that firmware is suitable only for use with the upstream/mainline kernels), you need to track down how it's getting pulled in.

**Reply by andregirard37** [reply]
So, I was adding the recipe linux-firmware , that I think is pulling poky/meta/recipes-kernel/linux-firmware/linux-firmware_20231030.bb in my image. Is that a no no and should be removed ? Not sure why this was added in the first place as I am not the original designer who added it. But without it, my image successfully builds. Is there a way to add it safely or is that linux-firmware recipe added anyway in the background without having to explicitly call it ? Thanks!!

**Reply by madisongh** [reply]
Well, it's going to pull in a huge pile of firmware that you probably don't need - the main `linux-firmware` package RRECOMMENDS all the other firmware packages that recipe creates.

If it turns out you do need it for some reason, you could try adding `linux-firmware-nvidia-tegra` to the `BAD_RECOMMENDATIONS` setting in your recipe (or add that setting if you don't already have one).

**Reply by madisongh** [reply]
If you have drivers that require specific firmware out of the `linux-firmware` bundle, it's generally better to include specific `RDEPENDS` on the subset of the firmware you need in the recipes that have such dependencies, or use the `MACHINE_xxx_RDEPENDS/RRECOMMENDS` settings in the machine config to get them included.
